### ISSUE TITLE
scrutiny empty dashboard

### ISSUE THREAD

**Issue by jcastro** [truenas/apps#882]
Hi!  
I just installed this app, and it doesn't seem to be working! I'm getting an empty dashboard, and the logs show the following:  

I also tried to check the documentation for community apps, but there's nothing about it.  

I'm running `ElectricEel-24.10.0.2` bare-metal on a Dell PowerEdge R730XD. I have also tried ticking the `Elevate Privileges` with same results

This is the output of `smartctl --scan``
```
/dev/sda -d scsi # /dev/sda, SCSI device
/dev/sdb -d scsi # /dev/sdb, SCSI device
/dev/sdc -d scsi # /dev/sdc, SCSI device
/dev/sdd -d scsi # /dev/sdd, SCSI device
/dev/sde -d scsi # /dev/sde, SCSI device
/dev/bus/0 -d megaraid,0 # /dev/bus/0 [megaraid_disk_00], SCSI device
/dev/bus/0 -d megaraid,2 # /dev/bus/0 [megaraid_disk_02], SCSI device
/dev/bus/0 -d megaraid,11 # /dev/bus/0 [megaraid_disk_11], SCSI device
/dev/bus/0 -d megaraid,12 # /dev/bus/0 [megaraid_disk_12], SCSI device
/dev/bus/0 -d megaraid,13 # /dev/bus/0 [megaraid_disk_13], SCSI device
/dev/nvme0 -d nvme # /dev/nvme0, NVMe device
/dev/nvme1 -d nvme # /dev/nvme1, NVMe device
```

Logs from the app
```
2024-11-11 13:38:03.029841+00:00ts=2024-11-11T13:38:03.029604Z lvl=info msg=Listening log_id=0snryz8W000 service=tcp-listener transport=http addr=:8086 port=8086
2024-11-11 13:38:07.783067+00:00scrutiny api not ready
2024-11-11 13:38:07.783429+00:00starting scrutiny
2024-11-11 13:38:07.794257+00:002024-11-11T13:38:07.794257448Z
2024-11-11 13:38:07.794290+00:00___   ___  ____  __  __  ____  ____  _  _  _  _
2024-11-11 13:38:07.794314+00:00/ __) / __)(  _ \(  )(  )(_  _)(_  _)( \( )( \/ )
2024-11-11 13:38:07.794324+00:00\__ \( (__  )   / )(__)(   )(   _)(_  )  (  \  /
2024-11-11 13:38:07.794332+00:00(___/ \___)(_)\_)(______) (__) (____)(_)\_) (__)
2024-11-11 13:38:07.794343+00:00github.com/AnalogJ/scrutiny                             dev-0.8.1
2024-11-11 13:38:07.794365+00:002024-11-11T13:38:07.794365621Z
2024-11-11 13:38:07.794374+00:00Start the scrutiny server
2024-11-11 13:38:07.794290+00:002024/11/11 13:38:07 No configuration file found at /opt/scrutiny/config/scrutiny.yaml. Using Defaults.
2024-11-11 13:38:07.794418+00:00time="2024-11-11T13:38:07Z" level=info msg="Trying to connect to scrutiny sqlite db: /opt/scrutiny/config/scrutiny.db\n" type=web
2024-11-11 13:38:07.795096+00:00time="2024-11-11T13:38:07Z" level=info msg="Successfully connected to scrutiny sqlite db: /opt/scrutiny/config/scrutiny.db\n" type=web
2024-11-11 13:38:07.795132+00:00time="2024-11-11T13:38:07Z" level=info msg="InfluxDB certificate verification: true\n" type=web
2024-11-11 13:38:07.872177+00:00time="2024-11-11T13:38:07Z" level=info msg="Database migration starting. Please wait, this process may take a long time...." type=web
2024-11-11 13:38:07.873093+00:00time="2024-11-11T13:38:07Z" level=info msg="Database migration completed successfully" type=web
2024-11-11 13:38:07.873122+00:00time="2024-11-11T13:38:07Z" level=info msg="SQLite global configuration migrations starting. Please wait...." type=web
2024-11-11 13:38:07.873314+00:00time="2024-11-11T13:38:07Z" level=info msg="SQLite global configuration migrations completed successfully" type=web
2024-11-11 13:38:12.691725+00:00time="2024-11-11T13:38:12Z" level=info msg="127.0.0.1 - truenas [11/Nov/2024:13:38:12 +0000] \"GET /health\" 200 7467 \"\" \"curl/7.88.1\" (1ms)" clientIP=127.0.0.1 hostname=truenas latency=1 method=GET path=/health referer= respLength=7467 statusCode=200 type=web userAgent=curl/7.88.1
2024-11-11 13:38:12.794165+00:00scrutiny api not ready
2024-11-11 13:38:17.808548+00:00scrutiny api not ready
2024-11-11 13:38:22.789982+00:00time="2024-11-11T13:38:22Z" level=info msg="127.0.0.1 - truenas [11/Nov/2024:13:38:22 +0000] \"GET /health\" 200 7467 \"\" \"curl/7.88.1\" (1ms)" clientIP=127.0.0.1 hostname=truenas latency=1 method=GET path=/health referer= respLength=7467 statusCode=200 type=web userAgent=curl/7.88.1
2024-11-11 13:38:22.819871+00:00scrutiny api not ready
```

<img width="1257" alt="image" src="https://github.com/user-attachments/assets/7c652e7b-043a-4a1d-8de6-683e02d9a695">

Thanks!

**Comment by stavros-k** [comment]
Did you add any disks in the disks section?

**Comment by jcastro** [comment]
> Did you add any disks in the disks section?

No! As I couldn't find any documentation from the app itself, I took a look at the Docker file in their repo where they say  
```
Scrutiny uses smartctl --scan to detect devices/drives.
```  
So I assumed the output of the command shows the disks, so they are added automatically?

**Comment by stavros-k** [comment]
> > Did you add any disks in the disks section?
> 
> No! As I couldn't find any documentation from the app itself, I took a look at the Docker file in their repo where they say
> 
> ```
> Scrutiny uses smartctl --scan to detect devices/drives.
> ```
> 
> So I assumed the output of the command shows the disks, so they are added automatically?

Not exactly. the `/run/udev` that is mounted gives some info, but in order for scrutiny to get more details you need to define the disks.

**Comment by jcastro** [comment]
> > > Did you add any disks in the disks section?
> > 
> > 
> > No! As I couldn't find any documentation from the app itself, I took a look at the Docker file in their repo where they say
> > ```
> > Scrutiny uses smartctl --scan to detect devices/drives.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > So I assumed the output of the command shows the disks, so they are added automatically?
> 
> Not exactly. the `/run/udev` that is mounted gives some info, but in order for scrutiny to get more details you need to define the disks.

Got it! I will give it a try. Just to make sure I need to add the '/dev/sda' on the 'Host disk path'
<img width="428" alt="image" src="https://github.com/user-attachments/assets/083b7a27-26e0-4c90-8436-a261e8034fa7">

**Comment by stavros-k** [comment]
> > > > Did you add any disks in the disks section?
> > > 
> > > 
> > > No! As I couldn't find any documentation from the app itself, I took a look at the Docker file in their repo where they say
> > > ```
> > > Scrutiny uses smartctl --scan to detect devices/drives.
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > So I assumed the output of the command shows the disks, so they are added automatically?
> > 
> > 
> > Not exactly. the `/run/udev` that is mounted gives some info, but in order for scrutiny to get more details you need to define the disks.
> 
> Got it! I will give it a try. Just to make sure I need to add the '/dev/sda' on the 'Host disk path' <img alt="image" width="428" src="https://private-user-images.githubusercontent.com/190036/384955147-083b7a27-26e0-4c90-8436-a261e8034fa7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzMzU1MTQsIm5iZiI6MTczMTMzNTIxNCwicGF0aCI6Ii8xOTAwMzYvMzg0OTU1MTQ3LTA4M2I3YTI3LTI2ZTAtNGM5MC04NDM2LWEyNjFlODAzNGZhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExMVQxNDI2NTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04MzBhNzIyMWNkZDQzOThmNmU2NDUxNGMxYjExN2M1MDhkN2Q3OTg1MDViZjA5ZmQ3MDA0NTNhNjU1Y2U0OGJmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZicdhcFsthufe_3d-aiQQn36cXit601txsW0Qj-tQPY">

On both fields you need to add `/dev/sda` (or whatever disks you have available).
The might be an error for the host path (notice that is limited to `/mnt`), but I'm changing this here https://github.com/truenas/apps/pull/883/commits/1851491a52c74e2715309992d1207b92c77dfd28

**Comment by jcastro** [comment]
Added two disks but still the dashboard is empty 🤔 
<img width="424" alt="image" src="https://github.com/user-attachments/assets/9a24803a-733a-4490-a06f-ccca2c9a02d0">
